### ISSUE TITLE
Precision-Recall Curve does not show class labels

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#601]
**Describe the bug**

When labels are passed into the `PrecisionRecallCurve` the visualization is not drawn correctly.

**To Reproduce**

```python
from yellowbrick.classifier import PrecisionRecallCurve 
from yellowbrick.dataset import load_game 

from sklearn.preprocessing import LabelEncoder 
from sklearn.naive_bayes import MultinomialNB 
from sklearn.model_selection import train_test_split as tts 

# Load the dataset and label encode the target 
data = load_game() 
X = data.iloc[:, data.columns != 'outcome']
y = LabelEncoder().fit_transform(data['outcome'])

# Create train test splits 
X_train, X_test, y_train, y_test = tts(X, y, test_size=0.2, shuffle=True)

oz = PrecisionRecallCurve(
    MultinomialNB(), per_class=True, iso_f1_curves=True, fill_area=False, 
    micro=False, classes=["loss", "draw", "win"]
)
oz.fit(X_train, y_train)
oz.score(X_test, y_test)
oz.poof()
```

**Dataset**

I used the game multi-label dataset from the UCI Machine Learning repository, as wrangled by the yellowbrick datasets module.

**Expected behavior**

When the target, `y` is label encoded (e.g. via the `LabelEncoder`) to classes 0, 1, and 2 and class names are passed in via the `classes` param of the visualizer, the legend should display the class names. However, in this case the visualization does not appear at all:

![figure_1](https://user-images.githubusercontent.com/745966/45109335-3c59a800-b10d-11e8-9386-e74de86282eb.png)

**Traceback**

No exception is raised, however, the following warning is issued:

```
/Users/benjamin/.pyenv/versions/3.6.2/envs/yellowbrick3/lib/python3.6/site-packages/numpy/lib/arraysetops.py:522: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  mask |= (ar1 == a)
/Users/benjamin/.pyenv/versions/3.6.2/envs/yellowbrick3/lib/python3.6/site-packages/sklearn/metrics/ranking.py:444: RuntimeWarning: invalid value encountered in true_divide
  recall = tps / tps[-1]
objc[46460]: Class FIFinderSyncExtensionHost is implemented in both /System/Library/PrivateFrameworks/FinderKit.framework/Versions/A/FinderKit (0x7fffb473cc90) and /System/Library/PrivateFrameworks/FileProvider.framework/OverrideBundles/FinderSyncCollaborationFileProviderOverride.bundle/Contents/MacOS/FinderSyncCollaborationFileProviderOverride (0x12814ecd8). One of the two will be used. Which one is undefined.
```

**Desktop (please complete the following information):**
 - OS: macOS High Sierra 10.13.6
 - Python Version 3.6.2
 - Yellowbrick Version 0.9 develop 

**Additional context**

This was a known bug during development but was left in as a rare case.

**Comment by zjpoh** [comment]
@bbengfort I would like to take a stab at this. Btw, what is the meaning of the hacktoberfest label?

**Comment by bbengfort** [comment]
@zjpoh that would be great - though you'll have to wait until #599 is merged before you can do a fix! Do you want to work with @rebeccabilbro in doing a code review of that PR?

The [hacktoberfest label](https://github.com/DistrictDataLabs/yellowbrick/issues?q=is%3Aopen+is%3Aissue+label%3Ahacktoberfest) is meant to designate issues that would be good for folks who are participating in [Hacktoberfest](https://hacktoberfest.digitalocean.com/), particularly [PyData Manchester folks](https://twitter.com/_JAStark/status/1037431605890048000).

**Comment by zjpoh** [comment]
@bbengfort Sure. I’ll take a look at that tonight.

**Comment by bbengfort** [comment]
@zjpoh still interested in fixing this bug?

**Comment by zjpoh** [comment]
@bbengfort Sorry again for being so irresponsive for such a long time. 

I think this error is due to `MultinomialNB` not `PrecisionRecallCurve` even though I haven't figured out the exact problem yet. By simply changing the model to `RidgeClassifier`, the correct plot is shown.

![prcurve](https://user-images.githubusercontent.com/8103276/50733427-1fc48f80-1142-11e9-8194-dfb9db47ef38.png)

Here is the code to reproduce the plots above.
```
from yellowbrick.classifier import PrecisionRecallCurve 
from yellowbrick.datasets import load_game 

from sklearn.preprocessing import LabelEncoder
from sklearn.linear_model import RidgeClassifier
from sklearn.model_selection import train_test_split as tts 

# Load the dataset and label encode the target 
X, y = load_game()
X = X.replace({"x": 0, "b": 1, "o": 2})
y = LabelEncoder().fit_transform(y)

# Create train test splits 
X_train, X_test, y_train, y_test = tts(X, y, test_size=0.2, shuffle=True)

oz = PrecisionRecallCurve(
    RidgeClassifier(), per_class=True, iso_f1_curves=True,
    fill_area=False, micro=False
)
oz.fit(X_train, y_train)
oz.score(X_test, y_test)
oz.poof()
```

**Comment by zjpoh** [comment]
@bbengfort Sorry for the long wait. Waiting for the final approval now. Should have it in a day or two.

**Comment by rebeccabilbro** [comment]
Thanks so much for sticking with this @zjpoh! @bbengfort is traveling for the next few days, in and out of stable wifi, but will be back at the end of this week -- we're both looking forward to resolving this bug; let me know if there's anything I can do in the interim!

**Comment by zjpoh** [comment]
https://github.com/scikit-learn/scikit-learn/issues/13043 addresses the problem of getting `nan` in the above example.
